app/actions.py

```diff
@@ -629,10 +629,40 @@ def cursorSelectPageUp(self):
     Returns:
       None.
     """
-    if self.selectionMode == app.selectable.kSelectionNone:
-      self.doSelectionMode(app.selectable.kSelectionCharacter)
+    if self.selectionMode != app.selectable.kSelectionCharacter:
+      self.selectionCharacter()
+    self.__cursorPageUp()
+
+  def cursorAltSelectPageDown(self):
+    """
+    DESCRIPTION HERE
+
+    Args:
+      None.
+
+    Returns:
+      None.
+    """
+    if self.selectionMode != app.selectable.kSelectionBlock:
+      self.selectionBlock()
+    self.__cursorPageDown()
+
+  def cursorAltSelectPageUp(self):
+    """
+    DESCRIPTION HERE
+
+    Args:
+      None.
+
+    Returns:
+      None.
+    """
+    if self.selectionMode != app.selectable.kSelectionBlock:
+      self.selectionBlock()
```

Comment:
If this call were inside an inner-loop (called frequently) then avoiding the function call can make a big performance difference. 

In this case the inner function has a 'if' to early out if it is being set to the same value. 

I'd prefer not to have the condition for these at this level so the code is cleaner to read.

(I like to have code be cleaner and easier to read, but there are times (lots of times) I will write weird code for the sake of performance. i.e. If a user may notice the difference I'll sacrifice how the code looks. In this case, I don't think a user will be able to notice the difference).

Reply:
Okay sounds good! I'll keep that in mind from now. I've been trying to write clean and easy to read code (like if you read it aloud, you would understand it without having to think too much about like some idoms in python such as return x == y or y == z etc). Still conflicted whether I should use these or not, or make it more clear through if statements. What's your take on this?